### PR TITLE
feat(kit): new pipe `HintErrorPipe`

### LINKED ISSUE #1399: 🐞 - A couple of problems with the new FieldErrorPipe
### Which @taiga-ui/* package(s) are the source of the bug?

kit

### Please provide a link to a minimal reproduction of the bug

_No response_

### 1) Difficulties in using with TuiHintDirective

I need to write strange extra code to pass FormControl's error into `tuiHint`:

```
<tui-input-number
    [formControl]="myFormControl"
    [tuiHint]="([] | tuiFieldError | async)?.message || ''"
>
</tui-input-number>
```

So...

`[] | tuiFieldError | async` - AsyncPipe returns type `TuiValidationError<{}> | null`. It incompatible with tuiHint's type `PolymorpheusContent<{}> | null`.

Okay, there is an object as a `TuiValidationError`, so I need to get value from the `message` field...
I need to use Optional Chaining to access the `message` field like this:

`([] | tuiFieldError | async)?.message` - Returns `PolymorpheusContent<{}> | undefined` and it's still incompatible with `tuiHint` type `PolymorpheusContent<{}> | null`.

And to ensure type compatibility, at the end I need to add "logical OR operator (||)" with an empty string. So, for each place I should write this:

`([] | tuiFieldError | async)?.message || ''`.

As I discussed with @waterplea, we can simplify this by creating another pipe, like `tuiErrorHint` which will immediately return some error text for `tuiHint`.

### 2) ExpressionChangedAfterItHasBeenCheckedError while using pipe with tuiHint directive

When we over the mouse on some input which has a `tuiHint`, and also when we start writing some new value, we have this ExpressionChangedAfterItHasBeenChecked error.

Video:
https://i.gyazo.com/75866e05cb40ff9ff03818ff242b6dd0.mp4

### Angular version

12.2.0

### Taiga UI version

2.34.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

Closes #1399 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/kit/pipes/hint-error/hint-error-pipe.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import {TuiFieldErrorPipe} from '../field-error';
+
+// @dynamic
+@Pipe({
+    name: 'tuiHintError',
+    pure: false,
+})
+export class TuiHintErrorPipe implements PipeTransform, OnDestroy {
+    asyncPipe!: AsyncPipe;
+    fieldErrorPipe!: TuiFieldErrorPipe;
```
Comment (splincode) [inline projects/kit/pipes/hint-error/hint-error-pipe.ts:14]: private?

File: projects/kit/pipes/hint-error/hint-error-pipe.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+@Pipe({
+    name: 'tuiHintError',
+    pure: false,
+})
+export class TuiHintErrorPipe implements PipeTransform, OnDestroy {
+    private readonly asyncPipe!: AsyncPipe;
+    private readonly fieldErrorPipe!: TuiFieldErrorPipe;
+
+    constructor(@Inject(INJECTOR) readonly injector: Injector) {
+        const localInjector = Injector.create({
```
Comment (waterplea) [inline projects/kit/pipes/hint-error/hint-error-pipe.ts:16]: Let's move this to private fields out of constructor. Also `injector` should be `private` and `INJECTOR` is only needed if you do not use static `Injector.create` which we do, so you might as well change it to `Injector`.

File: projects/kit/pipes/hint-error/hint-error-pipe.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            providers: [{provide: AsyncPipe}, {provide: TuiFieldErrorPipe}],
+            parent: injector,
+        });
+
+        this.asyncPipe = localInjector.get(AsyncPipe);
+        this.fieldErrorPipe = localInjector.get(TuiFieldErrorPipe);
+    }
+
+    transform(order: ReadonlyArray<string>): PolymorpheusContent | null {
```
Comment (waterplea) [inline projects/kit/pipes/hint-error/hint-error-pipe.ts:25]: Let's drop `null`

**Review by waterplea — COMMENTED** [review]
How about we rename it to `tuiFieldErrorContent` pipe and put it in the same module?

**Comment by vladimirpotekhin** [comment]
> How about we rename it to `tuiFieldErrorContent` pipe and put it in the same module?

Good idea 👍 
Renamed and moved in `TuiFieldErrorPipeModule`

**Review by waterplea — APPROVED** [review]

**Pull request merged**